tests/platform/test_reboot.py

```diff
@@ -0,0 +1,121 @@
+"""
+Check platform status after reboot. Three types of reboot are covered in this script:
+* Cold reboot
+* Fast reboot
+* Warm reboot
+
+This script is to cover the test case 'Reload configuration' in the SONiC platform test plan:
+https://github.com/Azure/SONiC/blob/master/doc/pmon/sonic_platform_test_plan.md
+"""
+import logging
+import re
+import os
+import time
+import sys
+
+import pytest
+
+from ansible_host import ansible_host
+from utilities import wait_until
+from check_critical_services import check_critical_services
+from check_interface_status import check_interface_status
+from check_transceiver_status import check_transceiver_basic
+from check_transceiver_status import all_transceivers_detected
+
+
+def reboot_and_check(localhost, dut, reboot_type="cold"):
+    """
+    Perform the specified type of reboot and check platform status.
+    """
+    dut.command("show platform summary")
+    lab_conn_graph_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), \
+        "../../ansible/files/lab_connection_graph.xml")
+    conn_graph_facts = localhost.conn_graph_facts(host=dut.hostname, filename=lab_conn_graph_file).\
+        contacted['localhost']['ansible_facts']
+    interfaces = conn_graph_facts["device_conn"]
+    asic_type = dut.shell("show platform summary | awk '/ASIC: / {print$2}'")["stdout"].strip()
+
+    logging.info("Run %s reboot on DUT" % reboot_type)
+    if reboot_type == "cold":
+        reboot_cmd = "sudo reboot"
+    elif reboot_type == "fast":
+        reboot_cmd = "sudo fast-reboot &"
+    elif reboot_type == "warm":
+        reboot_cmd = "sudo warm-reboot &"
+    else:
+        assert False, "Reboot type %s is not supported" % reboot_type
+    dut.shell(reboot_cmd)
+
+    logging.info("Wait for DUT to go down")
+    localhost.wait_for(host=dut.hostname, port=22, state="stopped", delay=10, timeout=300)
```

Comment:
i think it will be better to have different timeout for each type of boot. the values are not the same and can reduce the test execution as well.

Reply:
If all types of reboot work fine, the timeout would not affect execution time. But in case of rebooting issue, we need to fine tune the timeout value to reduce execution time. I'll add another commit to fine tune the timeout values. Thanks!

Reply:
New commit added.